### PR TITLE
docs: add documentation for the new 'first' operator

### PR DESCRIPTION
**Pull request by bothra90**
image

### CODE REVIEW

File: docs/pages/api-reference/operators.md [hunk 1]
```diff
@@ -101,6 +101,19 @@ The `aggregate` operator has the following parameters:
 
 <pre snippet="api-reference/operators_ref#aggregate"></pre>
 
+### Groupby/ First
```
Comment (nikhilgarg28) [inline docs/pages/api-reference/operators.md:104]: Please add a code snippet example?

File: docs/pages/api-reference/operators.md [hunk 2]
```diff
@@ -101,6 +101,19 @@ The `aggregate` operator has the following parameters:
...
+
+Fennel allows you to groupby and get the first entry (by timestamp) in each group using the `first` operator.
+The api requires you to chain the `groupby` operator with the `first` operator.
```
Comment (nikhilgarg28) [inline docs/pages/api-reference/operators.md:107]: 'API requires you to...' feels a bit weird from language POV. Instead, maybe simplify and say '`...first operator must always be preceeded by a groupby (else an error is raised)...`

File: docs/pages/api-reference/operators.md [hunk 3]
```diff
@@ -101,6 +101,19 @@ The `aggregate` operator has the following parameters:
...
+
+The `groupby` operator takes the following parameters:
+
+1. `*args: str` - positional arguments that specify the list of fields to group by.
```
Comment (nikhilgarg28) [inline docs/pages/api-reference/operators.md:111]: Not sure if this is needed here? Having this in two places makes it harder to keep the documentation upto date as the groupby operator evolves.

**Review by nikhilgarg28 — CHANGES_REQUESTED** [review]

File: docs/pages/api-reference/operators.md (hunk 3 above)
Comment (aditya-nambiar) [reply docs/pages/api-reference/operators.md:111]: Fyi we dont have a seperate groupby operator documentation, it is combined with aggregate

File: docs/pages/api-reference/operators.md (hunk 1 above)
Comment (bothra90) [reply docs/pages/api-reference/operators.md:104]: Missed that. Will add something

Comment (bothra90) [reply docs/pages/api-reference/operators.md:104]: Added an example. I am not able to visualize the docs on my local environment. If your setup is working, can you preview and make sure it looks good?

File: docs/pages/api-reference/operators.md (hunk 2 above)
Comment (bothra90) [reply docs/pages/api-reference/operators.md]: changed the wording. Let me know what you think now

File: docs/pages/api-reference/operators.md (hunk 3 above)
Comment (bothra90) [reply docs/pages/api-reference/operators.md]: +1 to what Nikhil is saying (about having documentaiton for groupby duplicated), but also `groupby in itself is not a full operator (e.g. we can't end a pipeline at `groupby`) which makes it challenging. Any ideas for this @nikhilgarg28 ?

**Review by nikhilgarg28 — APPROVED** [review]

**Pull request merged**